python/pyspark/serializers.py

```diff
@@ -182,6 +182,23 @@ def loads(self, obj):
         raise NotImplementedError
 
 
+class ArrowSerializer(FramedSerializer):
+    """
+    Serializes an Arrow stream.
+    """
+
+    def dumps(self, obj):
+        raise NotImplementedError
+
+    def loads(self, obj):
+        import pyarrow as pa
+        reader = pa.RecordBatchFileReader(pa.BufferReader(obj))
+        return reader.read_all()
```

Comment:
This will read all batches in a framed byte array from a stream and return.  The stream can have multiple framed byte arrays, so it repeats until end of stream.  

How many batches this reads depends on how it serialized.  When calling `toPandas()`, it collects all batches and then serializes each one to Python as an iterator.  So in this case, `reader.read_all()` will read 1 batch at a time.